qemu/tests/ple_test.py

```diff
@@ -0,0 +1,71 @@
+import re
+import logging
+
+from avocado.utils import cpu
+from virttest import env_process
+from virttest import error_context
+from virttest import arch
+from virttest.utils_kernel_module import KernelModuleHandler
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    ple test:
+    1) Check if ple is enabled on host, if not, enable it
+    2) Boot guest and run unixbench inside guest
+    3) Record benchmark scores and shutdown guest
+    4) Disable ple on host
+    5) Boot guest and run unixbench inside guest again
+    6) Compare benchmark scores with step 3)
+    7) Restore env, enable ple
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def run_unixbench(cmd):
+        """
+        Run unixbench inside guest, return benchmark scores
+        """
+        error_context.context("Run unixbench inside guest", logging.info)
+        output = session.cmd_output_safe(cmd, timeout=4800)
+        scores = re.findall(r"System Benchmarks Index Score\s+(\d+\.?\d+)",
+                            output)
+        return [float(i) for i in scores]
+
+    error_context.context("Enable ple on host", logging.info)
+    module = arch.get_kvm_module_list()[1]
+    moduleHandler = KernelModuleHandler(module)
+    mod_param = params["mod_param"] % params["ple_value"]
+    moduleHandler.reload_module(force=True, params=mod_param)
+
+    host_cpu_count = cpu.online_count()
+    params["vcpu_maxcpus"] = host_cpu_count
```

Comment:
Suggest to make `params["vcpu_maxcpus"] = params["smp"]`

Reply:
Any particular reason?

Reply:
Upstream use `smp` as the main, but we use `vcpu_maxcpus`, so need to make them all equal to host_cpu_count.

Reply:
@PaulYuuu  It is me that suggest to remove the assignment for `smp` in the pre comment, framework can calculate it automatically, same with `vcpu_maxcpus` for this case.
Could you explain more for adding it back ?

Reply:
VT uses `smp` by default, and the value is 1. If we only reset vcpu_maxcpus here, other users will launch the guest with `-smp 1,maxcpus=${vcpu_maxcpus}`

Reply:
Got it, thanks for your explanation.